compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -546,26 +546,26 @@ public void generateVarStore(MethodVisitor mv, BIRNode.BIRVariableDcl varDcl, in
         }
     }
 
-    private BType getSmallerUnsignedIntSubType(BType lhsType, BType rhsType) {
+    private BType getPossibleUnsignedIntSubType(BType lhsType, BType rhsType) {
```

Comment:
How about something like
```suggestion
    private BType getSmallestBuiltInIntSubTypeContainingTypes(BType lhsType, BType rhsType) {
```

"Possible" doesn't quite imply what we are returning/checking for, right?

Reply:
It should be largest. How about `getLargestBuiltInIntSubTypeContainingTypes` ?

Reply:
+1 to go with smallest since it is the process of selecting `smallest K`

Reply:
By containing types I meant containing lhsType and rhsType.

If lhsType is int:Unsigned8 and rhsType is int:Unsigned16, 
a. int:Unsigned8 contains int:Unsigned8 but not int:Unsigned16
b. int:Unsigned16 contains int:Unsigned8 and int:Unsigned16
c. int:Unsigned32 contains int:Unsigned8 and int:Unsigned16

So a is out anyway. Of b and c we select the smallest.

Maybe we can add in the Unsigned bit too
```suggestion
    private BType getSmallestBuiltInUnsignedIntSubTypeContainingTypes(BType lhsType, BType rhsType) {
```